### ISSUE TITLE
[SDK 48 Beta] Android - Build fails on EAS

### ISSUE THREAD

**Issue by dmahajan980** [expo/expo#21152]
### Summary

After upgrading to SDK 48 (beta), I am getting the following build error for Android:

```
> Task :react-native-gradle-plugin:pluginDescriptors
> Task :react-native-gradle-plugin:processResources
> Task :react-native-gradle-plugin:compileKotlin
> Task :react-native-gradle-plugin:compileJava NO-SOURCE
> Task :react-native-gradle-plugin:classes
> Task :react-native-gradle-plugin:inspectClassesForKotlinIC
> Task :react-native-gradle-plugin:jar
[stderr] FAILURE: Build completed with 2 failures.
[stderr] 1: Task failed with an exception.
[stderr] -----------
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/android/app/build.gradle' line: 21
[stderr] * What went wrong:
[stderr] A problem occurred evaluating project ':app'.
[stderr] > path may not be null or empty string. path=''
[stderr] * Try:
[stderr] >
[stderr] Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] 2: Task failed with an exception.
[stderr] -----------
[stderr] * What went wrong:
[stderr] A problem occurred configuring project ':app'.
[stderr] > compileSdkVersion is not specified. Please add it to build.gradle
[stderr] * Try:
[stderr] > Run with --stacktrace option to get the stack trace.
[stderr] > Run with --info or --debug option to get more log output.
[stderr] > Run with --scan to get full insights.
[stderr] ==============================================================================
[stderr] * Get more help at https://help.gradle.org
[stderr] BUILD FAILED in 1m 33s
5 actionable tasks: 5 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

This build was for a custom dev client.

### What platform(s) does this occur on?

Android

### Environment

I'm not sure the local details matter since the build failed on EAS server. Anyway, here's the output:

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.6
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 19.4.0 - /usr/local/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.2.0 - /usr/local/bin/npm
      Watchman: 2022.12.19.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/lib/ruby/gems/3.1.0/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
      Android SDK:
        API Levels: 30, 31, 32, 33
        Build Tools: 28.0.3, 30.0.2, 30.0.3, 31.0.0, 33.0.0
        System Images: android-33 | Google APIs ARM 64 v8a
        Android NDK: 25.1.8937393
    IDEs:
      Android Studio: 2021.2 AI-212.5712.43.2112.8815526
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^48.0.0-beta => 48.0.0-beta.2 
      react: 18.2.0 => 18.2.0 
      react-dom: ^18.2.0 => 18.2.0 
      react-native: 0.71.2 => 0.71.2 
      react-native-web: ^0.18.10 => 0.18.10 
    npmGlobalPackages:
      eas-cli: 3.5.2
      expo-cli: 6.0.8
    Expo Workflow: managed
```

### Minimal reproducible example

https://github.com/dmahajan980/eas-build-failure-reproducible

If it helps, the EAS Build ID is `04d36d34-3341-4406-8b81-4d394eebcf22`. Happy to help with any other form of reproduction.

**Comment by brentvatne** [comment]
hi there! please try to isolate to a minimal reproducible example :) this means a repository that we can clone and run. thank you!

**Comment by dmahajan980** [comment]
Hi @brentvatne! Apologies for not attaching a reproducible earlier. I have created a new project with the same packages and EAS config as the original one. Please find it here:

https://github.com/dmahajan980/eas-build-failure-reproducible

To create a new build, run the command: 

```
eas build staging:dev:simulator
```

This is the same reproducible attached in #21151 since both issues have occured in the same project. Please let me know if anything else is needed. Also, since the reproducible is available, can you also re-open this issue?

Thanks

**Comment by brentvatne** [comment]
@dmahajan980 - the project builds for me on android after running `npx expo install --fix`. make sure you run `npx expo install --fix` to align deps.

**Comment by dmahajan980** [comment]
Hi @brentvatne! Thanks for helping out on this one. I was still facing errors while building the app but upgrading `react-native-mmkv` fixed the issue.

After you mentioned the reproducible built for you, I tried building my project and the reproducible separately. The reproducible built successfully while the project build failed. I suspect has to do with different versions of one or more packages pinned in yarn.lock file since I only copied the dependencies and necessary dev dependencies from my app and generated a new lock file in the reproducible.

Also, this time the failure reason was different than what I shared in the issue's description. For the most recent failure, seems that `react-native-mmkv` was the culprit. It only [supports React Native 0.71 from version `2.6.1`](https://github.com/mrousavy/react-native-mmkv/releases/tag/v2.6.1). Upgrading that to 2.6.1 fixed the issue.
